test/test-dynamic-project-scope.ts

```
@@ -0,0 +1,344 @@
+  const result = await client.callTool("get_project", { project_id: projectId });
+  assert.ok(result.content, "Should have content");
+  const content = result.content[0];
+  assert.ok("text" in content, "Content should have text");
+  const project = JSON.parse(content.text as string);
+  return project.id.toString();
+}
+
+describe("Dynamic project scope", { concurrency: 1 }, () => {
+  describe("with GITLAB_ALLOWED_PROJECT_IDS set", () => {
+    let mockGitLab: MockGitLabServer;
+    let servers: ServerInstance[] = [];
```

ESLint

[error] 'servers' is never reassigned. Use 'const' instead.

(prefer-const)